src/libraries/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -442,13 +477,20 @@ public int LastIndexOf(string value, int startIndex, int count, StringComparison
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
 
-            // Special case for 0 length input strings
-            if (this.Length == 0 && (startIndex == -1 || startIndex == 0))
-                return (value.Length == 0) ? 0 : -1;
+            // Make sure we're not out of range.
+            // If 'this' is empty, treat startIndex = -1 the same as startIndex = 0.
 
-            // Now after handling empty strings, make sure we're not out of range
-            if (startIndex < 0 || startIndex > this.Length)
-                throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_Index);
+            if ((uint)startIndex > (uint)this.Length)
+            {
+                if (startIndex == -1 && this.Length == 0)
+                {
```

Comment:
I am not sure if this is correct but this something you want to double check. here is the scenario 

this = ""
value = "A"
startIndex = -1

I believe this should return -1 but now we'll return 0 which is wrong I guess.